src/Talis/Persona/Client/Base.php

```diff
@@ -345,21 +344,30 @@ protected function performRequest($url, array $opts)
         try {
             $response = $request->send();
         } catch(RequestException $exception) {
-            if (isset($exception->hasResponse) && $exception->hasResponse) {
-                $status = $exception->getResponse()->getStatusCode();
-            } else {
+            $response = $exception->getRequest()->getResponse();
+            if(isset($response))
```

Comment:
When I debugged this - the 404 scenario throws the `ClientErrorResponseException` subclass. This has no `hasResponse` method but it does have `getResponse`. Given this, the approach above felt cleaner.